### PR TITLE
refact(es6): use es6 syntax.

### LINKED ISSUE #10: remove "merge-descriptors" dependency at all?
Hi -

this time no bug, only an idea:
Since there is Node v4.0 out, would it make sense to replace the line
```javascript
merge(resources[locale], resource);
```
in index.js, line 62 with 
```javascript
Object.assign(resources[locale], resource);
```
?

### PR DESCRIPTION
**Pull request by hotoo**
* fixed #10, But iojs-2 not support Object.assign, so use object-assign shim instead for now.
* use es6 syntax.
* use eslint to instead jshint.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,7 +53,7 @@ Patch locales functions to koa app.
 - {Application} app: koa app instance.
 - {Object} options: optional params
   - {String} functionName: locale function name patch on koa context. Optional, default is `__`.
-  - {String} dir: locales resources store directory. Optional, default is `$PWD/locales`.
```
Comment (hotoo) [inline README.md]: @Deprecated Use options.dirs instead.

**Comment by hotoo** [comment]
@popomore @fengmk2

**Comment by fengmk2** [comment]
+1

**Comment by fengmk2** [comment]
1.4.2

**Pull request merged**